tools/gonilnilfunctions/pkg/analyzer/testdata/funcs.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package testdata
+
+func ReturnReturnOkay() (any, error) {
+	var i interface{}
+	return i, nil
+}
+
+func OneGoodOneBad() (any, error) { // want "Function OneGoodOneBad can return an error, and has a statement that returns only nils"
+	var i interface{}
+	if true {
+		return i, nil
+	}
+	return nil, nil
+}
+
+func OneBadOneGood() (any, error) { // want "Function OneBadOneGood can return an error, and has a statement that returns only nils"
+	var i interface{}
+	if true {
+		return nil, nil
+	}
+	return i, nil
+}
+
+func EmptyFunc() {}
+
+func TwoNilNils() (any, error) { // want "Function TwoNilNils can return an error, and has a statement that returns only nils"
+	if true {
+		return nil, nil
+	}
+	return nil, nil
+}
+
+func ThreeArgs() (any, any, error) { // want "Function ThreeArgs can return an error, and has a statement that returns only nils"
```

Comment:
Yeah, I thought about this on implementation. It was easy enough to make this generic to more arguments, and I figured a developer might have similar conceptions about `thing1, thing2, error` function returns (where you often expect _either_ `thing1` + `thing2` _or_ `error`, so my default inclination was to extend this to any number of arguments, where one is an error.

I'm happy to restrict this to only two return result functions, but I figured the same logic usually applies to more-than-two-return-value functions. More-than-two-return-value functions are (and usually should be) rarer, in either case.

Reply:
If we only care about `func() (something, error)`, I wonder whether it would be simpler to use semgrep? It seems like the benefit of an AST-level check is that it can handle an arbitrary number of returned values.

Reply:
> If we only care about `func() (something, error)`, I wonder whether it would be simpler to use semgrep? It seems like the benefit of an AST-level check is that it can handle an arbitrary number of returned values.

I don't think we can create a semgrep rule that only looks at newly introduced code, can we?